### DISCUSSION TITLE
Asking about running Flask-SocketIO on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by pvtoan** [miguelgrinberg/Flask-SocketIO#2050] [Q&A]
Hi,

Actually, I can normally run the example code from your page "https://pypi.org/project/Flask-SocketIO/" on Windows 11. After running, it can show me the local host link like "http://127.0.0.1:5000"

However, when I run the same code on Ubuntu 22.04, it shows nothing.

The version of flask and flask-socketio is the same on both Windows and Ubuntu as follows
flask 3.0.2                   
flask-socketio 5.3.6   

Do you know what my problem is and how to fix that?

**Comment by miguelgrinberg** [comment]
You need to be more specific.

- What application are you running
- What command do you run in your terminal
- What is the exact output you get in your terminal
- A picture of your browser
- Any output in the console log of the browser

**Comment by pvtoan** [comment]
Hi,

1. My app is to exchange data including real-time parameters and images between python flask and web app (written in html and javascript).

2. I press directly the "Run" button in Pycharm or VSCode to run my *.py file.

3. Here is the output I got from my terminal when I run on Windows 11
 * Serving Flask app 'test'
 * Debug mode: off
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000
Press CTRL+C to quit

Meanwhile, when I run exact the same code on Ubuntu 22.04, my *.py can be run without showing anything compared with the above output on Windows 11.

4. When I click to the link above "http://127.0.0.1:5000", my browser is shown as the attached picture
![browser](https://github.com/miguelgrinberg/Flask-SocketIO/assets/119550856/f13fc106-f485-4ae0-bfee-a9faf35580bd)

Note that, in my "index.html", I just have one simple command to show the header "Testing" as shown in the picture.

5. I attach the 2nd picture to show console log I viewed from Windows 11 (if that is what you mentioned).
![consolelogs](https://github.com/miguelgrinberg/Flask-SocketIO/assets/119550856/0d924243-9ca5-4fbd-9373-3e349a8c2689)


Please let me know if you need more info for your reference!

**Comment by miguelgrinberg** [comment]
If you run `pip freeze` in the Windows and Linux machines do you get identical output? Make sure that you activate your virtual environment before running `pip freeze`.

**Comment by pvtoan** [comment]
Hi, 

I run pip freeze on both Windows and Linux with my environment "myenv1" and they have the same 
flask 3.0.2
flask-socketio 5.3.6

However, when I run the code on my another environment "myenv2" (flask 3.0.2, flask-socketio 5.3.6), and it works normally.

Thus, I think there might be some problem with flask-related packages installed on "myenv1" on Linux.

Do you have any idea what packages related to flask, flask-socketio I should pay attention?
The reason is that I have many other packages that I don't want to change version because of compatibility issue with other packages

**Reply by miguelgrinberg** [reply]
Use [pipdeptree](https://pypi.org/project/pipdeptree/) to list your packages organized as a tree of dependencies.

**Comment by pvtoan** [comment]
Thank you for your help!

However, I cannot figure out the solution. Thus, I just recreated a new environment to use.
